pmlc/dialect/pxa/transforms/reorder_layouts.cc

```diff
@@ -1549,14 +1594,33 @@ mlir::Optional<ReorderDesc> optimizeLayoutForReads(
     MemoryUsageDesc &memoryDesc,
     mlir::DenseMap<mlir::Value, mlir::AffineMap> &memLayoutMaps,
     bool makeUserLayoutsExplicit) {
+  /*
+   * There are two ways of determining the target layouts for a tensor
+   * 1. If a tensor appears in an Op such as Convolution, then we would like
+   * to set the target layouts to the known optimal blocked layouts
+   * E.g., if the input tensor is NCHW then we would like the target layout to
+   * be NCHWc16
+   *
+   * 2. If a tensor appears in a generic op not recognized by our pattern
+   * matchers such as Convolution pattern matchers, then we would like to derive
+   * optimal layouts from first principles.
+   */
+
+  /* If the flag - makeUserLayoutsExplicit is set, then we adopt strategy 1).
+   * If the said flag is not set, then we go for strategy 2)
+   */
```

Comment:
Use C++ style comments (avoid C style comments, i.e. /* */

Reply:
Yes. For the subsequent comments I added, I used "//".